table/block_based_table_reader.cc

```diff
@@ -173,6 +173,25 @@ Cache::Handle* GetEntryFromCache(Cache* block_cache, const Slice& key,
   return cache_handle;
 }
 
+bool PrefixExtractorChanged(std::string prefix_extractor_block,
+                            const SliceTransform* prefix_extractor,
+                            BlockBasedTableOptions::IndexType index_type) {
+  bool prefix_extractor_changed = false;
+  // only needed for hash based index
+  if (index_type != BlockBasedTableOptions::kHashSearch &&
```

Comment:
I hope we do none of the checking below when `prefix_extractor == nullptr`

Reply:
I think it's OK that if prefix_extractor == nullptr we treat that the user has disabled prefix bloom and we don't do any check.